File: Chatroom_java/app/src/main/java/com/example/chatroom_java/data/LoadData.java
package com.example.chatroom_java.data;

import android.content.Context;

import com.example.chatroom_java.R;

import org.json.JSONArray;
import org.json.JSONObject;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class LoadData {
    public static String loadJSONFromAsset(Context context, String fileName) {
        String json = "";
        try {
            InputStream inputStream = context.getAssets().open(fileName);
            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
            StringBuilder stringBuilder = new StringBuilder();
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                stringBuilder.append(line);
            }
            inputStream.close();
            json = stringBuilder.toString();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return json;
    }

    public static List<Chat> parseChatJSON(String json) {
        List<Chat> chatList = new ArrayList<>();
        try {
            JSONArray jsonArray = new JSONArray(json);
            for (int i = 0; i < jsonArray.length(); i++) {
                JSONObject jsonObject = jsonArray.getJSONObject(i);
                String name = jsonObject.getString("name");
                String text = jsonObject.getString("text");
                int image = getImageForName(name); // Assign image based on name
                Chat chat = new Chat((long) i, name, image, text);
                chatList.add(chat);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return chatList;
    }

    private static int getImageForName(String name) {
        switch (name) {
            case "User 1":
                return R.drawable.user_image1;
            case "User 2":
                return R.drawable.user_image2;
            case "User 3":
                return R.drawable.user_image3;
            case "User 4":
                return R.drawable.user_image4;
            case "User 5":
                return R.drawable.user_image5;
            default:
                return R.drawable.user_image;
        }

    }
}

File: Chatroom_hackthon/app/src/main/java/com/example/chatroom_hackthon/data/LoadData.kt
package com.example.chatroom_hackthon.data

import android.content.Context
import org.json.JSONArray
import java.io.InputStream


fun loadJSONFromAsset(context:Context,fileName: String): String {
    val json: String
    try {
        val inputStream: InputStream = context.assets.open("$fileName")
        json = inputStream.bufferedReader().use { it.readText() }
    } catch (e: Exception) {
        e.printStackTrace()
        return ""
    }
    return json
}
// 解析 JSON 数据并返回 Chat 对象列表
fun parseChatJSON(json: String): List<Chat> {
    val chatList = mutableListOf<Chat>()
    try {
        val jsonArray = JSONArray(json)
        for (i in 0 until jsonArray.length()) {
            val jsonObject = jsonArray.getJSONObject(i)
            val name = jsonObject.getString("name")
            val text = jsonObject.getString("text")
            val image = getImageForName(name) // Assign image based on name
            val chat = Chat(i.toLong(), name, image, text)
            chatList.add(chat)
        }
    } catch (e: Exception) {
        e.printStackTrace()
    }
    return chatList
}